src/migrations/20220115000000-audit-system.js

```diff
@@ -0,0 +1,921 @@
+/*
+Postgresql has a make table/trigger/function name length of 63 bytes.
+In order to more likely stay under this limit all names are abbreviated where possible.
+Statically defined names are known to be short enough, but are still abbreviated to keep
+  consistency.
+All dynamically generated table/trigger/function are abbreviated where possible.
+
+The following list are the abbreviations used:
+Z - Prefix to keep all audit tables, triggers, & function at the end of any list
+A - Audit
+L - Log
+F - Function
+T - Trigger
+
+Examples:
+  ZAL - Z Audit Log
+  ZALF - Z Audit Log Function
+  ZALT - Z Audit Log Trigger
+
+All table/type/function/trigger create/alter/drops are monitored and logged to a ddl audit log.
+All calls to truncate a table are logged into the ddl audit log for the corresponding table.
+All methods to create a table will trigger the creation of a corresponding set of audit
+  table/function/triggers necessary to audit all activity on the triggering table.
+When a table is dropped the functions and triggers monitoring that table are removed, the audit
+   log it self is not removed to maintain audit integrity.
+Truncate and Update to the audit log tables is not permitted through the used of triggers.
+Delete on audit tables is only allowed when the 'ARCHIVE AUDIT LOG' descriptor is passed within
+  the transaction. This is to allow stale records to be removed only under archival process.
+  All other attempts to delete from an audit log table are prevented.
+All insert/update/deletes performed on a monitored table will be logged.
+Updates will only store the fields that change as part of the specific call.
+Inserts and Deletes will keep the whole record.
+Specific fields across all tables or for a specific table can be flagged as not being of
+  significance requiring a audit log entry if they are the only field to change. If any other
+  field also changes, all field that have changes will be recorded.
+A passed in user id will be recorded with any log generated, If no user id was passed, -1 is used.
+A passed in transaction id will be recorded with any log generated, if no transaction id was passed,
+  then the built in postgresql transaction id will be used. The same transaction id should be used
+  for all db calls generated form a single UI generated event to aid in associating all
+  corresponding changes.
+A passed in session sig will be recorded with any log generated, if no session sig is passed, null
+  will be used.
+An optional descriptor can be passed and will be recorded with any log generated if supplied.
+*/
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => queryInterface.sequelize.transaction(
+    async (transaction) => {
+      let syncronizer;
```

Comment:
I was having race issues without forcing it to wait for the await.

Reply:
Should be able to `await` without assigning the result to a variable

Reply:
I was having a race condition issue with several of the calls, forcing each to wait for output has corrected this.

Reply:
I could see running into an issue when there was mixed usage between `await` and chaining (when exceptions were caught with `.catch` instead of `try/catch`) but there should be no difference between assigning the result of what is being `await`ed and not assigning. The `await` will wait for the promise to be resolved regardless of if the result is assigned to a variable or ignored.